aws-sdk-core/lib/aws-sdk-core/plugins/sqs_md5s.rb

```diff
@@ -0,0 +1,148 @@
+require 'openssl'
+
+module Aws
+  module Plugins
+
+    # @seahorse.client.option [Boolean] :verify_checksums (true)
+    #   When `true` MD5 checksums will be computed for messages sent to
+    #   an SQS queue and matched against MD5 checksums returned by Amazon SQS.
+    #   `Aws::Errors::Checksum` errors are raised for cases where checksums do
+    #   not match.
+    class SQSMd5s < Seahorse::Client::Plugin
+      OPERATIONS_TO_VERIFY = [:send_message, :send_message_batch]
+
+      # @api private
+      class Handler < Seahorse::Client::Handler
+        def call(context)
+          @handler.call(context).on_success do |response|
+            case context.operation_name
+            when :send_message
+              validate_send_message(context, response)
+            when :send_message_batch
+              validate_send_message_batch(context, response)
+            end
+          end
+        end
+
+        private
+
+        TRANSPORT_TYPE_ENCODINGS = {
+          'String' => 1,
+          'Binary' => 2,
+          'Number' => 1
+        }
+
+        NORMALIZED_ENCODING = Encoding::UTF_8
+
+        def validate_send_message(context, response)
+          body = context.params[:message_body]
+          attributes = context.params[:message_attributes]
+          validate_single_message(body, attributes, response)
+        end
+
+        def validate_send_message_batch(context, response)
+          context.params[:entries].each do |entry|
+            id = entry[:id]
+            body = entry[:message_body]
+            attributes = entry[:message_attributes]
+            message_response = response.successful.select { |r| r.id == id }[0]
+            unless message_response.nil?
+              validate_single_message(body, attributes, message_response)
```

Comment:
@trevorrowe This is the line you mentioned in the previous commit: I simple added a `nil` check. Only messages which are which are placed in the `successful` array are verified.